### PR TITLE
Remove duplicated config item Domain to fix bug

### PR DESCRIPTION
**Pull request by lunny**
Remove duplicated config item Domain and only use ROOT_URL and fix go get when port is not 80 or 443

### CODE REVIEW

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 1]
```diff
@@ -82,9 +82,8 @@ Values containing `#` or `;` must be quoted using `` ` `` or `"""`.
 
 ## Server (`server`)
 
-- `PROTOCOL`: **http**: \[http, https, fcgi, unix\]
-- `DOMAIN`: **localhost**: Domain name of this server.
 - `ROOT_URL`: **%(PROTOCOL)s://%(DOMAIN)s:%(HTTP\_PORT)s/**: Full public URL of Gitea server.
```
Comment (sapk) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md:85]: If we break config we should do it completely rigth and maybe rename variable. 
Like using more self-explaining name PUBLIC_URL=https://try.gitea.io and LISTEN_ADDR=unix:///var/run/gitea.sock, LISTEN_ADDR=https://0.0.0.0:443

A breaking note should follow but that way admin know what is the impact of config. 
Maybe we can also try to migrate the config.

Comment (lunny) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md:85]: @sapk I like your idea. But this maybe break the old config.

Comment (sapk) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md:85]: Your change will break thing in the (rare ?) case if ROOT_URL is not defined in the config file.

Comment (lunny) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md:85]: I will send another commit to this PR.

**Comment by thehowI** [comment]
This does not fix any bug. It is just a waste of time.

**Comment by thehowl** [comment]
just to be clear, @thehowi is a copycat and not actually me.

**Pull request closed without merging**